qiskit_nature/second_q/hamiltonians/lattices/kagome_lattice.py

```diff
@@ -0,0 +1,305 @@
+            )
+        return list_of_edges
+
+    def _default_position(self) -> Dict[int, List[float]]:
+        """Return a dictionary of default positions for visualization of a two-dimensional lattice.
+
+        Returns:
+            Dict[int, List[float]] : The keys are the labels of lattice points,
+                and the values are two-dimensional coordinates.
+        """
+        size = self.size
+        boundary_condition = self.boundary_condition
+        num_sites_per_cell = self.num_sites_per_cell
+        pos = {}
+        width = 0.0
+        if boundary_condition == BoundaryCondition.PERIODIC:
+            # the positions are shifted along the x- and y-direction
+            # when the boundary condition is periodic.
+            # The width of the shift is fixed to 0.2.
+            width = 0.2
+        for cell_idx in range(np.prod(size)):
+            # maps an cell index to two-dimensional coordinate
+            # the positions are shifted so that the edges between boundaries can be seen
+            # for the periodic cases.
+            cell_coord = np.array(divmod(cell_idx, size[0])[::-1]) + width * np.cos(
+                np.pi * (np.array(divmod(cell_idx, size[0]))) / (np.array(size)[::-1] - 1)
+            )
+
+            for i in range(3):
+                node_i = num_sites_per_cell * cell_idx + i
+                pos[node_i] = (np.dot(cell_coord, self.basis) + self.cell_positions[i]).tolist()
+        return pos
+
+    def __init__(
+        self,
+        rows: int,
+        cols: int,
+        edge_parameter: complex = 1.0,
+        onsite_parameter: complex = 0.0,
+        boundary_condition: BoundaryCondition = BoundaryCondition.OPEN,
+    ) -> None:
+        """
+        Args:
+            rows: Length of the x direction.
+            cols: Length of the y direction.
+            edge_parameter: Weight on all the edges, specified as a single value
+                Defaults to 1.0,
+            onsite_parameter: Weight on the self-loops, which are edges connecting a node to itself.
+                Defaults to 0.0.
+            boundary_condition: Boundary condition for the lattice.
+                The available boundary conditions are:
+                BoundaryCondition.OPEN, BoundaryCondition.PERIODIC.
+                Defaults to BoundaryCondition.OPEN.
+
+        Raises:
+            ValueError: Given size, edge parameter or boundary condition are invalid values.
+        """
+        self.rows = rows
+        self.cols = cols
+        self.size = (rows, cols)
+        self.dim = 2
+        self.boundary_condition = boundary_condition
+        self.num_sites_per_cell = 3
+        self.cell_positions = np.array([[0, 0], [1, 0], [1 / 2, np.sqrt(3) / 2]])
+        self.basis = np.array([[2, 0], [1, np.sqrt(3)]])
+
+        self.edge_parameter = edge_parameter
+        self.onsite_parameter = onsite_parameter
```

Comment:
These cannot be simple attributes unless setting them also has the intended effects on the underlying graph. In the other classes all of these are private attributes for that reason and then we only expose the relevant ones as "read-only" `@property` attributes